request.go

```diff
@@ -35,10 +35,50 @@ func fillInsert(enc *encoder, spaceNo uint32, tuple interface{}) error {
 	return enc.Encode(tuple)
 }
 
-func fillSelect(enc *encoder, spaceNo, indexNo, offset, limit, iterator uint32, key interface{}) error {
-	enc.EncodeMapLen(6)
-	fillIterator(enc, offset, limit, iterator)
-	return fillSearch(enc, spaceNo, indexNo, key)
+func fillSelect(enc *encoder, spaceNo, indexNo, offset, limit, iterator uint32,
+	key, after interface{}, fetchPos bool) error {
+	mapLen := 6
+	if fetchPos {
+		mapLen += 1
+	}
+	if after != nil {
+		mapLen += 1
+	}
+	if err := enc.EncodeMapLen(mapLen); err != nil {
+		return err
+	}
+	if err := fillIterator(enc, offset, limit, iterator); err != nil {
+		return err
+	}
+	if err := fillSearch(enc, spaceNo, indexNo, key); err != nil {
+		return err
+	}
+	if fetchPos {
+		if err := encodeUint(enc, KeyFetchPos); err != nil {
+			return err
+		}
+		if err := enc.EncodeBool(fetchPos); err != nil {
+			return err
+		}
+	}
+	if after != nil {
+		if pos, ok := after.([]byte); ok {
+			if err := encodeUint(enc, KeyAfterPos); err != nil {
+				return err
+			}
+			if err := enc.EncodeString(string(pos)); err != nil {
```

Comment:
What's vmihailenco/msgpack rules for MP_BIN?

Reply:
Tarantool throw decoding error when I try to use `EncodeBinary` (it uses `encode_str`/`decode_str` in fact).

Reply:
Well, it's kinda sad, but it doesn't seems that we will be able to do anything about it